gluten-delta/src/main/scala/io/glutenproject/extension/DeltaRewriteTransformerRules.scala

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.extension
+
+import io.glutenproject.execution.{FileSourceScanExecTransformer, ProjectExecTransformer}
+import io.glutenproject.extension.DeltaRewriteTransformerRules.columnMappingRule
+import io.glutenproject.extension.columnar.TransformHints
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.expressions.{Alias, Attribute, AttributeReference}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreeNodeTag
+import org.apache.spark.sql.delta.{DeltaParquetFileFormat, NoMapping}
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.datasources.FileFormat
+import org.apache.spark.sql.types.{StructField, StructType}
+
+import scala.collection._
+
+class DeltaRewriteTransformerRules extends RewriteTransformerRules {
+  override def rules: Seq[Rule[SparkPlan]] = columnMappingRule :: Nil
+}
+
+object DeltaRewriteTransformerRules {
+
+  private val COLUMN_MAPPING_RULE_TAG: TreeNodeTag[String] =
+    TreeNodeTag[String]("io.glutenproject.delta.column.mapping")
+
+  private def notAppliedColumnMappingRule(plan: SparkPlan): Boolean = {
+    plan.getTagValue(COLUMN_MAPPING_RULE_TAG).isEmpty
```

Comment:
What's the logic here? Seems `COLUMN_MAPPING_RULE_TAG` is not empty at initialization?

Reply:
Firstly, this `COLUMN_MAPPING_RULE_TAG` is used to avoid a delta scan applies this rule multiple times.
At initialization, the original transformer can't be tagged, so `COLUMN_MAPPING_RULE_TAG` is empty.